client/src/components/GrantRequired.js

```diff
@@ -0,0 +1,37 @@
+import React from 'react'
+import ResourceAddNoGrantRequestForm from 'components/ResourceAddNoGrantRequestForm'
+import { Loader } from 'components/Loader'
+import { useUser } from 'hooks/useUser'
+
+import { Modal } from 'components/Modal'
+
+export default ({ children }) => {
+  const { isAbleToAddResources } = useUser()
+  const [userOk, setUserOk] = React.useState()
+
+  React.useEffect(() => {
+    const asyncCheck = async () => {
+      setUserOk(await isAbleToAddResources())
+    }
+    if (userOk === undefined) asyncCheck()
+  })
+
+  if (userOk === undefined) return <Loader />
+
+  // show feature request form if user does not have a grant
+  if (!userOk) {
+    return (
+      <>
+        {children}
+        <Modal showing nondismissable>
+          <ResourceAddNoGrantRequestForm
+            portalId="5187852"
+            formId="960eaa6e-dc9a-4921-874f-40ba9bae8583"
```

Comment:
Are these magic strings from Hubspot?

Reply:
Oh I see it now. Maybe leave a comment or something? Hubspot isn't mentioned in this file anywhere.

Reply:
ok i moved them to env vars